### DISCUSSION TITLE
[Enhancement] [upload] upload作为表单组件，没有校验功能/ElUpload cannot validated in ElForm

### DISCUSSION THREAD

**Discussion by Leonw26** [element-plus/element-plus#10918] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `upload`

### Features / 特性
1. The ElUpload component is expected to be validated in the ElForm / 期望ElUpload组件在ElForm里可以验证

### demo
```vue
<script setup>
import { ref } from 'vue'
import { ElForm, ElFormItem, ElUpload } from 'element-plus';

const form = ref({url: ''})

const rules = {
  url: {requires: true, message: "请上传图片"}
}
</script>

<template>
  <ElForm :model="form" :rules="rules">
    <ElFormItem prop="url">
      <ElUpload v-model="form.url" />
    </ElFormItem>
  </ElForm>
</template>
```

### Additional comments

期望是在提交表单前，可以利用ElForm的表单校验功能拦截提交。

**Comment by arkfancy** [comment]
我是这样实现的 对el-upload包装一个modelValue（直接用v-model:file-list不确定行不行）
大概这样，其它还有些业务校验的省略了
<template>
 <el-upload v-model:file-list="fileList“ />
</template>
<script>
const props = defineProps({
  modelValue: Object
})
const emtis = defineEmits(update:modelValue)
const fileList = ref([])
watch(() => props.modelValue,() => {
  fileList.value = props.modelValue
})
watch(fileList, () => {
  emits("update:modelValue", fileList.value);
},deep)
</script>
